### PR TITLE
Change U calculation to match deepmind pseudocode

### PR DESCRIPTION
**Pull request by Tilps**
While investigating why fpu mixing was having a stronger effect for leela lite than for lc0 I noticed that leela lite's formula for U was slightly different to the formula in lc0.

I'm still of the opinion that the lc0 formula is an accurate rendering of the formulas presented in papers, but I went back to the deepmind pseudocode published with the alphazero paper and found that they too had the same difference in their calculation of U.

I haven't got any positive tests in favour of this change as of yet, so it doesn't seem super relevant - but maybe it'll help someone testing fpu mixing to see if it has an amplifying effect together.

### CODE REVIEW

**Comment by mooskagh** [comment]
If we do this change, we can also remove GetChildrenVisits.
Actually GetChildrenVisits did show up as non-zero in profiler (because of condition probably), so having it removed will also improve nps by a (very) tiny bit.

**Comment by Tilps** [comment]
GetChildrenVisits is still used in the logic for creating training data as its the number of visits to normalize by.

**Comment by Tilps** [comment]
(I still have no improvement to go with this - but I have decided that logically this change probably should go with a slight decrease in cpuct - so I am going to try testing whether it responds better to a lower cpuct than default.)

**Comment by Tilps** [comment]
There was an interim result posted in off-topic which suggests that this change has a positive delta when baseline and comparison use fpu absolute rather than our default fpu strategy.  I might give that a try after my current tests.

**Comment by Tilps** [comment]
Preliminary results suggest baseline is reacting more positive to cpuct 2.7 compared to 3.0 - compared to the alternative U code.  Still running.

**Comment by Tilps** [comment]
Based on my ongoing test I think that what is happening is that this new U value changes the ideal tune - and that at the nodes I am testing at, our default is clearly not optimal for either.  Without deriving ideal tune for both before and after and then comparing them, I don't think I can come to a clear conclusion whether this change is good at all.

**Comment by Ttl** [comment]
Some more test games with self-play settings:

```
Score of lc0_pr699 vs lc0_master: 200 - 181 - 619  [0.509] 1000
Elo difference: 6.60 +/- 13.28, LOS: 83.48 %, DrawRatio: 61.9 %
```

Network 40626, Parameters: `--threads=1 --fpu-strategy=absolute --smart-pruning-factor=0 --minibatch-size=32 -cache-history-length=7 --policy-softmax-temp=1.0 --cpuct=2.5`.

File: src/mcts/search.cc [hunk 1]
```diff
@@ -207,8 +207,7 @@ std::vector<std::string> Search::GetVerboseStats(Node* node,
                                                  bool is_black_to_move) const {
   const float fpu = GetFpu(params_, node, node == root_node_);
   const float cpuct = ComputeCpuct(params_, node->GetN());
-  const float U_coeff =
-      cpuct * std::sqrt(std::max(node->GetChildrenVisits(), 1u));
```
Comment (mooskagh) [inline src/mcts/search.cc]: I think now we can delete GetChildrenVisits() completely.

Comment (Tilps) [reply src/mcts/search.cc]: Its still used in GetV4TrainingData.

Comment (mooskagh) [reply src/mcts/search.cc]: It can be `GetN() - 1` there probably. :)

**Comment by Naphthalin** [comment]
In [node.h](https://github.com/LeelaChessZero/lc0/blob/master/src/mcts/node.h#L360) the comment should be changed as well for this PR.

**Comment by Tilps** [comment]
Looks to me that the comment is already accurate for after this change and misleading for before? :P

**Comment by Naphthalin** [comment]
Oh, good catch :D pretty sure though that wasn't intended and should anyway be differentiated if we introduce a flag there.

File: src/mcts/params.cc [hunk 2]
```diff
@@ -60,6 +60,10 @@ const OptionId SearchParams::kCpuctBaseId{
     "higher growth of Cpuct as number of node visits grows."};
 const OptionId SearchParams::kCpuctFactorId{
     "cpuct-factor", "CPuctFactor", "Multiplier for the cpuct growth formula."};
+const OptionId SearchParams::kNewUEnabledId{
+    "new-u-enabled", "NewUEnabled",
```
Comment (mooskagh) [inline src/mcts/params.cc]: Neither flag name, nor description (what corresponds to true and what to false?) are completely clear.

Maybe `--parent-based-u`? `--include-initial-visit-to-u`? `--u-parent-visits`?

**Comment by Tilps** [comment]
After reading 913 I expect to rollback the last patch which added the command line flag.

**Comment by Naphthalin** [comment]
If I understand this correctly, this behavior was never changed, so the PR might still be relevant.

**Comment by Tilps** [comment]
yeah its just so low priority that I've never come back to it.
